src/server/pps/server/api_server.go

```diff
@@ -874,41 +863,46 @@ func (a *apiServer) FlushPipelineJob(request *pps.FlushPipelineJobRequest, resp
 	defer func(start time.Time) {
 		a.Log(request, fmt.Sprintf("stream containing %d PipelineJobInfos", sent), retErr, time.Since(start))
 	}(time.Now())
-	var toRepos []*pfs.Repo
-	for _, pipeline := range request.ToPipelines {
-		toRepos = append(toRepos, client.NewRepo(pipeline.Name))
-	}
-
-	pachClient := a.env.GetPachClient(resp.Context())
-	return pachClient.FlushCommit(request.Commits, toRepos, func(ci *pfs.CommitInfo) error {
-		var pjis []*pps.PipelineJobInfo
-		// FlushPipelineJob passes -1 for history because we don't know which version
-		// of the pipeline created the output commit.
-		if err := a.listPipelineJob(resp.Context(), nil, ci.Commit, nil, -1, false, "", func(pji *pps.PipelineJobInfo) error {
-			pjis = append(pjis, pji)
-			return nil
-		}); err != nil {
-			return err
-		}
-		if len(pjis) == 0 {
-			// This is possible because the commit may be part of the stats
-			// branch of a pipeline, in which case it's not the output commit
-			// of any job, thus we ignore it, the job will be returned in
-			// another call to this function, the one for the job's output
-			// commit.
-			return nil
-		}
-		if len(pjis) > 1 {
-			return errors.Errorf("found too many jobs (%d) for output commit: %s", len(pjis), pfsdb.CommitKey(ci.Commit))
-		}
-		// Even though the commit has been finished the job isn't necessarily
-		// finished yet, so we block on its state as well.
-		pji, err := a.InspectPipelineJob(resp.Context(), &pps.InspectPipelineJobRequest{PipelineJob: pjis[0].PipelineJob, BlockState: true})
-		if err != nil {
-			return err
-		}
-		return resp.Send(pji)
-	})
+
+	// TODO(global ids): implement whatever use case this is trying to solve (maybe pfs.FlushJob can do most of the work?)
```

Comment:
With global ids this should be mostly doable with pfs.FlushJob or whatever we call it. The purpose of this function is basically to get back the jobs a commit caused so you can get logs from them / see why they failed.